test/micro/org/openjdk/bench/java/lang/constant/ReferenceClassDescResolve.java

```diff
@@ -0,0 +1,140 @@
+
+/**
+ * Compare array clone with equivalent System.arraycopy-based routines
+ */
+@BenchmarkMode(Mode.Throughput)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@Warmup(iterations = 3, time = 2)
+@Measurement(iterations = 6, time = 1)
+@Fork(1)
+@State(Scope.Benchmark)
+public class ReferenceClassDescResolve {
+    private static final MethodHandles.Lookup LOOKUP = MethodHandles.lookup();
+
+    static Class<?> resolve(MethodHandles.Lookup lookup, ClassDesc cd) throws ReflectiveOperationException {
+        if (cd.isPrimitive()) {
+            return (Class<?>) cd.resolveConstantDesc(lookup);
+        }
+        return resolveReference(lookup, cd);
+    }
+
+    static Class<?> resolveReference(MethodHandles.Lookup lookup, ClassDesc cd) throws ReflectiveOperationException {
+        if (cd.isArray()) {
+            var desc = cd.descriptorString();
+            if (desc.charAt(desc.length() - 1) != ';') {
+                // Primitive arrays
+                return lookup.findClass(desc);
+            }
+            int depth = 0;
+            while (desc.charAt(depth) == '[') {
+                depth++;
+            }
+            Class<?> clazz = lookup.findClass(desc.substring(depth + 1, desc.length() - 1).replace('/', '.'));
+            for (int i = 0; i < depth; i++)
+                clazz = clazz.arrayType();
+            return clazz;
+        }
+        var desc = cd.descriptorString();
+        return lookup.findClass(desc.substring(1, desc.length() - 1).replace('/', '.'));
+    }
+
+    public enum Kind {
+        CLASS(String.class),
+        CLASS_ARRAY(Integer[][].class),
+        PRIMITIVE_ARRAY(long[][][].class);
+
+        final ClassDesc desc;
+
+        Kind(Class<?> clz) {
+            this.desc = clz.describeConstable().orElseThrow();
+        }
+    }
+
+    public enum Implementation {
+        OLD {
+            @Override
+            Class<?> call(MethodHandles.Lookup lookup, ClassDesc cd) throws ReflectiveOperationException {
+                ClassDesc c = cd;
+                var desc = c.descriptorString();
+                int depth = 0;
+                while (desc.charAt(depth) == '[') {
+                    depth++;
+                }
+                for (int i = 0; i < depth; i++)
+                    c = c.componentType();
+
+                if (c.isPrimitive())
+                    return lookup.findClass(desc);
+                else {
+                    desc = c.descriptorString();
+                    Class<?> clazz = lookup.findClass(desc.substring(1, desc.length() - 1).replace('/', '.'));
+                    for (int i = 0; i < depth; i++)
+                        clazz = clazz.arrayType();
+                    return clazz;
+                }
+            }
+        },
+        NEW {
+            @Override
+            Class<?> call(MethodHandles.Lookup lookup, ClassDesc cd) throws ReflectiveOperationException {
+                return resolveReference(lookup, cd);
+            }
+        },
```

Comment:
I think this benchmark doesn't need this case as that's the implementation of `resolveConstantDesc`.   Was it there when you developed this fix?   Same for MethodTypeDesc benchmark.